src/classes/VOL_SharedCode.cls

```diff
@@ -400,66 +400,67 @@ global with sharing class VOL_SharedCode {
                 strSoql += strComma + strF;
                 strComma = ', ';
             }
-        }        
+        }
         strSoql += ' from Contact ';
         string strAnd = ' where ';
-        
+
         // make sure their settings haven't been completely cleared
         if (VolunteersSettings.Contact_Matching_Rule__c == null || VolunteersSettings.Contact_Matching_Rule__c == '') {
             VolunteersSettings.Contact_Matching_Rule__c = 'Firstname;Lastname;Email';
         }
-        
-        if (isLastnameInContactMatchRules) {
-            strSoql += strAnd + ' (Lastname=\'' + StrEscape(contact.Lastname) + '\') ';
+
+        final String rule = VolunteersSettings.Contact_Matching_Rule__c;

```

Comment:
I'm curious about the reasoning behind using `final` here. It works, I'd just like to learn more about the choice to use it. :)

Reply:
It's definitely not necessary but for me it says this is like a constant and should not be changed but only used to provide information.

Reply:
I didn't see that the settings were getting committed anywhere, but I'm not sure concerned about it either way.